File: src/solar/solar-common/util/lang/VmUtil.ts
export default class VmUtil {


    /**
     * 地图文件里是否拥有某资源（某文件）
     * @param resourcePath
     */
    static hasResource(resourcePath: string): boolean {
        let [file] = io.open(resourcePath, "r")
        if (file) {
            return true;
        }
        return false;
    }

    /**
     * 从地图文件里读取字符串内容
     * @param strResourcePath
     */
    static getResourceAsString(strResourcePath: string): string {
        let [file] = io.open(strResourcePath, "rb");
        if (!file) {
            return
        }
        let str = file.read("*a" as any) as string;
        file.close();
        return str;
    }


    /**
     * 从地图文件里读取Json内容
     * @param jsonResourcePath
     */
    static getResourceAsJson(jsonResourcePath: string): any {
        return JSON.parse(VmUtil.getResourceAsString(jsonResourcePath));
    }


    /**
     * 写文件
     * @param filename
     * @param content
     */
    static writeFile(filename: string, content: string) {
        let [file] = io.open(filename, "wb");
        file.write(content)
        file.close()
    }

}

File: scripts/solar/solar-common/util/lang/VmUtil.lua
local ____lualib = require("lualib_bundle")
local __TS__Class = ____lualib.__TS__Class
local __TS__SourceMapTraceBack = ____lualib.__TS__SourceMapTraceBack
__TS__SourceMapTraceBack(debug.getinfo(1).short_src, {["6"] = 1,["7"] = 1,["8"] = 1,["10"] = 1,["11"] = 8,["12"] = 9,["13"] = 10,["14"] = 11,["16"] = 13,["17"] = 8,["18"] = 20,["19"] = 21,["20"] = 22,["23"] = 25,["24"] = 26,["25"] = 27,["26"] = 20,["27"] = 35,["28"] = 36,["29"] = 35,["30"] = 45,["31"] = 46,["32"] = 47,["33"] = 48,["34"] = 45});
local ____exports = {}
____exports.default = __TS__Class()
local VmUtil = ____exports.default
VmUtil.name = "VmUtil"
function VmUtil.prototype.____constructor(self)
end
function VmUtil.hasResource(self, resourcePath)
    local file = io.open(resourcePath, "r")
    if file then
        return true
    end
    return false
end
function VmUtil.getResourceAsString(self, strResourcePath)
    local file = io.open(strResourcePath, "rb")
    if not file then
        return
    end
    local str = file:read("*a")
    file:close()
    return str
end
function VmUtil.getResourceAsJson(self, jsonResourcePath)
    return JSON:parse(____exports.default:getResourceAsString(jsonResourcePath))
end
function VmUtil.writeFile(self, filename, content)
    local file = io.open(filename, "wb")
    file:write(content)
    file:close()
end
return ____exports
